stingray/events.py

```diff
@@ -246,14 +265,59 @@ def to_lc(self, dt, tstart=None, tseg=None):
         -------
         lc: :class:`stingray.Lightcurve` object
         """
-        if tstart is None and self.gti is not None:
-            tstart = self.gti[0][0]
-            tseg = self.gti[-1][1] - tstart
-
         return Lightcurve.make_lightcurve(
-            self.time, dt, tstart=tstart, gti=self.gti, tseg=tseg, mjdref=self.mjdref
+            self.time, dt, tstart=tstart, gti=self._gti, tseg=tseg, mjdref=self.mjdref
         )
 
+    def to_timeseries(self, dt, array_attrs=None):
+        """Convert the event list to a :class:`stingray.StingrayTimeseries` object.
+
+        The result will be something similar to a light curve, but with arbitrary
+        attributes corresponding to a weighted sum of each specified attribute of
+        the event list.
```

Comment:
I'm not sure I understand here what you mean by "weighted" sum. If you have an EventList with an energy for each photon event, and you turn that into a `TimeSeries` object, should you not still have an array with time stamps and an array with energy values?